### DISCUSSION TITLE
Is it possible to add a pendingComponent for layout?

### DISCUSSION THREAD

**Discussion by brunoti** [TanStack/router#2054] [Q&A]
I have a case where my app has a defaultPending component. When I add a _layout route I want to change the defaultPendingComponent for routes using the layout.

Is that possible? I've tried setting the pendingComponent on the layout and tried to find stuff on the docs but couldn't find it.

**Comment by schiller-manuel** [comment]
so you would want a child route to inherit its parent's `pendingComponent` if it has not defined one itself?

**Reply by chinanderm** [reply]
Yes! Having this type of logic would be really great. My use case being different pending UIs based on where the user is in the route tree. Example: If the whole layout is loading, I would like a certain pending UI, versus if a deeply nested child is pending.

I understand this is probably difficult because you have to codesplit out pendingComponent from the rest of the route/layout. Maybe that's already being done for some things?
